inference-engine/src/multi_device/multi_device_exec_network.hpp

```diff
@@ -56,15 +57,40 @@ class ThreadSafeQueue {
             return false;
         }
     }
-
-    bool empty() {
+protected:
+    std::queue<T>   _queue;
+    std::mutex      _mutex;
+};
+template <typename T>
+class ThreadSafeBoundedQueue {
+public:
+    ThreadSafeBoundedQueue() = default;
+    bool try_push(T value) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (_capacity) {
+            _queue.push(std::move(value));
+        }
+        return _capacity;
+    }
+    bool try_pop(T& value) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (_capacity && !_queue.empty()) {
+            value = std::move(_queue.front());
+            _queue.pop();
+            return true;
+        } else {
+            return false;
+        }
+    }
+    void set_capacity(std::size_t newCapacity) {
         std::lock_guard<std::mutex> lock(_mutex);
-        return _queue.empty();
+        _capacity = newCapacity;
```

Comment:
`_capacity` is bool, `newCapacity` is size_t. Is it expected?

Reply:
Yes in this model we need just two states of queue enabled/disabled.